src/main/java/com/commercetools/pspadapter/paymentHandler/impl/PaymentHandler.java

```diff
@@ -343,22 +354,26 @@ private AddInterfaceInteraction createAddInterfaceInteractionAction(@Nonnull Pay
     private CompletionStage<PaymentHandleResponse> saveResponseToInterfaceInteraction(@Nullable String paymentId,
                                                                                       @Nullable String paymentIdType,
                                                                                       @Nonnull PayPalRESTException restException) {
-        CompletionStage<String> ctpPaymentIdStage = CompletableFuture.completedFuture(paymentId);
+        CompletionStage<Optional<String>> ctpPaymentIdOptStage = CompletableFuture.completedFuture(Optional.ofNullable(paymentId));
```

Comment:
But I always put `Opt` extension in the end of the object that is optional, so it's more consistent like this. If CompletionStage is optional, it would be named `ctpPaymentIdStageOpt`

Reply:
Ok, looks like you just weren't so consistent before ;)
https://github.com/commercetools/commercetools-paypal-plus-integration/blob/f1b227be7c504b2a4ea1b1b5a0f9987ad173cc14/src/main/java/com/commercetools/service/ctp/impl/CartServiceImpl.java#L65-L65

https://github.com/commercetools/commercetools-paypal-plus-integration/blob/1cf4423cbc28b0bebb9042bd5fcc7b4b16ffed4d/src/main/java/com/commercetools/service/ctp/impl/PaymentServiceImpl.java#L61-L61

Reply:
Yes, I was not consistent before 😄  https://github.com/commercetools/commercetools-paypal-plus-integration/pull/74/commits/82c08913b40251eec46547a42d0c521fb4b303e6